examples/generative/gaugan.py

```diff
@@ -0,0 +1,792 @@
+from. The generator of GauGAN takes as inputs the latents sampled from the Gaussian
+distribution as well as the one-hot encoded semantic segmentation label maps. The cue
+images act as style images that guide the generator to stylistic generation. This
+variational formulation helps GauGAN achieve diversity as well as fidelity.
+- **Multi-scale patch discriminator** : Inspired by the
+[PatchGAN](https://paperswithcode.com/method/patchgan) model,
+GauGAN uses a discriminator that assesses a given image on a patch basis
+and produces an averaged score.
+
+As we proceed with the example, we will discuss further details of the different
+components involved in our implementation.
+
+For a thorough review of GauGAN, please refer to
+[this article](https://blog.paperspace.com/nvidia-gaugan-introduction/).
+You are also encouraged to check
+[the official GauGAN website](https://nvlabs.github.io/SPADE/) that
+has many creative applications of GauGAN. This example assumes that the reader is already
+familiar with the fundamental concepts of GANs. If you need a refresher the following
+resources might be useful:
+
+* [Chapter on GANs](https://livebook.manning.com/book/deep-learning-with-python/chapter-8)
+from the Deep Learning with Python book by François Chollet.
+* GAN implementations on keras.io:
+    * [Data efficient GANs](https://keras.io/examples/generative/gan_ada)
+    * [CycleGAN](https://keras.io/examples/generative/cyclegan)
+    * [Conditional GAN](https://keras.io/examples/generative/conditional_gan)
+"""
+
+"""
+## Data collection
+
+We will be using the
+[Facades dataset](https://cmp.felk.cvut.cz/~tylecr1/facade/)
+for training our GauGAN model. Let's first download it and also install
+TensorFlow Addons.
+"""
+
+"""shell
+!gdown https://drive.google.com/uc?id=1q4FEjQg1YSb4mPx2VdxL7LXKYu3voTMj
+!unzip -q facades_data.zip
+!pip install -qqq tensorflow_addons
+"""
+
+"""
+## Imports
+"""
+
+import os
+import random
+import numpy as np
+from tqdm import tqdm
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from glob import glob
+from PIL import Image
+
+"""
+## Data splitting
+"""
+
+PATH = "./facades_data/"
+SPLIT = 0.2
+
+files = glob(PATH + "*.jpg")
+np.random.shuffle(files)
+
+split_index = int(len(files) * (1 - SPLIT))
+train_files = files[:split_index]
+val_files = files[split_index:]
+
+print(f"Total samples: {len(files)}.")
+print(f"Total training samples: {len(train_files)}.")
+print(f"Total validation samples: {len(val_files)}.")
+
+"""
+## Data loader
+"""
+
+BATCH_SIZE = 4
+IMG_HEIGHT = IMG_WIDTH = 256
+N_CLASS = 12
+AUTOTUNE = tf.data.AUTOTUNE
+
+
+def load(image_files, batch_size, is_train=True):
+    def _random_crop(
+        image,
+        segmentation_map,
+        labels,
+        crop_size=tf.convert_to_tensor((IMG_HEIGHT, IMG_WIDTH)),
```

Comment:
Avoid calling a function in a signature definition, so move the `convert_to_tensor` inside the function body